src/unitxt/metrics.py

```diff
@@ -534,6 +543,7 @@ def process(self, stream: Stream, stream_name: Optional[str] = None) -> Generato
 
         for instance in instances:
             instance["score"]["global"].update(global_score)
+            self.validate_ci_in_global_score(instance["score"]["global"])
```

Comment:
Perhaps combine here the `.update(global_score)` with the new validation into one method ?

Reply:
It would be better, but I still want to avoid these special if conditions.   What if we change
![image](https://github.com/user-attachments/assets/192746c5-634b-44ff-96a4-af94e5a2ce5c)
So the result will be initialized to the scores with 'nan'?

Reply:
Or even simpler, create a ci variable with default (nan,nan) and then update the result in any case?

Reply:
So you agree with my comment here: 
https://github.com/IBM/unitxt/pull/1065#discussion_r1694245144
?

I think that the global score might be overloaded. 
You wrote a special test "agaist" ci_scores that should not show in the global score..

Reply:
In addition, I think that `nan` should be kept to some error in the calculation (like too few cases, or too many samples yielding nan global score).
If at all, perhaps start with something that indicates `not calculated`.  Perhaps a string value as is, `"not calculated"`, rather than a float.

Reply:
I see your point.  I agree we e should not have nan if it was have nan when it's not calculate.  

So how about replacing

        confidence_interval = self.compute_global_confidence_intervals(
            references, predictions, task_data, score_name
        )
        global_score.update(confidence_interval)

with
 
        confidence_interval = self.compute_global_confidence_intervals(
            references, predictions, task_data, score_name
        ) 
        update_confidence_interval(global_score, confidence_interval)  
      
      
 The update_confidence_interval would check if confidence_interval is {}, and if so, it would delete score_ci_low and score_ci_high (and update the global score otherwise with confidence_interval).
 
 
This would not require checking after the fact if the fields exists.

Sorry if I'm focusing on it, but unitxt code is complex as is, and I want to localize complex change.

Reply:
No problem at all! Certainly nothing to be sorry about!

The issue is that we need to clean something that is already sitting in the global score (in each `instance["score"]["global"]`) . We do not see the whole problem just by looking at the newly computed `global_score` (the one just computed for the current metric, i.e., the one to be **updated into** the  `instance["score"]["global"]`-s).  We can suspect there might be a problem if the current metric does not compute CI, but we are not sure. We need to look back..

What you suggested in the last comment (https://github.com/IBM/unitxt/pull/1065#discussion_r1694255037) : 
`The update_confidence_interval would check if confidence_interval is {}, and if so, it would delete score_ci_low and score_ci_high (and update the global score otherwise with confidence_interval).`
means do look into `instance["score"]["global"]`, and delete if needed.
So I repeat my suggestion: to implement exactly what you say here, in one unified method of update-and-delete-whats-needed (https://github.com/IBM/unitxt/pull/1065#discussion_r1694243720)

Reply:
Ok. Got it. We need to delete instance["score"]["global"]  and not  from the global_score, which is just the current one.

LEt's go with what you suggest.

Reply:
Thanks @yoavkatz , just pushed a unified method. Turns out that with the current `global_score` at hand (in addition to `instance["score"]["global]`), the check (which you rightfully described as complex) is dramatically simpler. So your instincts led us in the right direction, and I am glad that you insisted!
I also wrote a long explanation..